lib/util/eyeglass_modules.js

```diff
@@ -0,0 +1,283 @@
+"use strict";
+
+var resolve = require("./resolve");
+var packageUtils = require("./package");
+var path = require("path");
+var merge = require("lodash.merge");
+var includes = require("lodash.includes");
+var semver = require("semver");
+var uuid = require("uuid");
+
+var EYEGLASS_KEYWORD = "eyeglass-module";
+var ROOT_ID = ":root-" + uuid.v4();
+
+function EyeglassModules(dir) {
+  dir = packageUtils.findNearestPackage(path.resolve(dir));
+
+  var pkg = packageUtils.getPackage(dir);
+
+  var moduleTree = resolveModule(dir, true);
+
+  //console.log("[moduleTree]", moduleTree);
+
+  var collection = dedupeModules(flattenModules(moduleTree));
+
+  this.collection = collection;
+  this.list = Object.keys(collection).map(function(name) {
+    return collection[name];
+  });
+  this.tree = pruneModuleTree.call(this, moduleTree);
+  this.projectName = moduleTree.name;
+}
+
+EyeglassModules.prototype.access = function(name, location) {
+  var mod = getFinalModule.call(this, name);
+
+  if (mod && !canAccessModule.call(this, name, location)) {
+    throw new Error("Could not import " + name + " from " + location);
+  }
+
+  return mod;
+};
+
+EyeglassModules.prototype.find = function(name) {
+  return getFinalModule.call(this, name);
+};
+
+EyeglassModules.prototype.findByPath = function(dir) {
+  return this.list.find(function(mod) {
+    if (mod.path === dir) {
+      return getFinalModule.call(this, mod.name);
+    }
+  }.bind(this));
+};
+
+// this is mostly for debugging purposes
+EyeglassModules.prototype.getGraph = function() {
+  var hierarchy = getHierarchy(this.tree);
+  hierarchy.label = getDecoratedRootName.call(this);
+  return require("archy")(hierarchy);
+};
+
+function getDecoratedRootName() {
```

Comment:
Semantics. Is the project moving towards `.call` to create pseudo private scope?